frontend/packages/dev-console/src/components/monitoring/alerts/monitoring-alerts-utils.tsx

```diff
@@ -145,11 +151,41 @@ export const monitoringAlertRows = (
   return rows;
 };
 
+export const getFilteredRows = (_filters, rowFilters, objects) => {
+  if (_.isEmpty(_filters)) {
+    return objects;
+  }
+
+  const allTableFilters = getAllTableFilters(rowFilters);
+  let filteredObjects = objects;
+  _.each(_filters, (value, name) => {
+    const filter = allTableFilters[name];
+    if (_.isFunction(filter)) {
+      if (name === 'alert-state') {
+        filteredObjects = _.filter(filteredObjects, (o) =>
+          !_.isEmpty(value.selected)
+            ? _.filter(o.alerts, (a) => filter(value, a)).length > 0 ||
+              (value.selected.has('inactive') && _.isEmpty(o.alerts))
+            : o,
+        );
+      } else {
+        filteredObjects = _.filter(filteredObjects, (o) => filter(value, o));
+      }
+    }
+  });
+
+  return filteredObjects;
+};
+
+const alertStateReducer = (rule: Rule) => {
+  return _.map(rule.alerts, (a) => a.state)[0] || 'inactive';
```

Comment:
Why only take the state of the first alert? Don't we need to iterate through all alerts looking for any alert that has a matching state?

Reply:
Or if not, you could just write rule.alerts[0]?.state, but I think @kyoto question here is valid

Reply:
I am mapping through all the alerts of the rule and it returns an array of a single element that is why I am taking the first element.

Reply:
But why mapping first over all elements and converting alerts to states, and then picking the first state. You could also pick the first alert state directly, or?

```js
  return rule.alerts[0].state || 'inactive';
```

I did not understand if we could reduce the alerts in any way instead of picking just the first one. /cc @kyoto

Reply:
It looks to me like this won't work when a rule has multiple active alerts with different states. For example, `_.map(rule.alerts, (a) => a.state)` could return `['firing', 'pending']`, and then this would ignore the `pending` alert.

Reply:
Great final solution with `inMatch`. Nice work @vikram-raj and @kyoto